modules/ismp/core/src/handlers/response.rs

```diff
@@ -33,6 +35,12 @@ where
 {
 	let signer = msg.signer.clone();
 
+	let data = sp_io::hashing::keccak_256(&msg.datagram.encode());
+	let relayer_public_key = Signature::decode(&mut signer.as_slice())
```

Comment:
Don't verify the signature, implement a method on the Signature object that just returns public key or address from the enum and use that as the signer everywhere msg.signer is  used.